pkg/watchdogd/watchdogd.go

```diff
@@ -7,40 +7,63 @@
 //
 // It starts in the running+armed state:
 //
-//             \| watchdogd Running     | watchdogd Stopped
+//              | watchdogd Running     | watchdogd Stopped
 //     ---------+-----------------------+--------------------------
 //     Watchdog | watchdogd is actively | machine will soon reboot
 //     Armed    | keeping machine alive |
 //     ---------+-----------------------+--------------------------
 //     Watchdog | a hang will not       | a hang will not reboot
 //     Disarmed | reboot the machine    | the machine
 //
-// The following signals control changing state:
-//
-//     - STOP: running -> stopped
-//     - CONT: stopped -> running
-//     - USR1: armed -> disarmed
-//     - USR2: disarmed -> armed
+
 package watchdogd
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"io/ioutil"
 	"log"
+	"net"
 	"os"
-	"os/signal"
-	"path/filepath"
-	"strconv"
 	"strings"
 	"time"
 
 	"github.com/u-root/u-root/pkg/watchdog"
 	"golang.org/x/sys/unix"
 )
 
-// DaemonProcess is the name of the daemon process.
-const DaemonProcess = "watchdogd"
+// UDS is the name of daemon's unix domain socket.
+const UDS = "/tmp/watchdogd"
+
+const (
+	OpStop     = iota // 0 - Stop the watchdogd petting.
+	OpContinue        // 1 - Continue the watchdogd petting.
+	OpDisarm          // 2 - Disarm the watchdog.
+	OpArm             // 3 - Arm the watchdog.
+)
+
+const (
+	OpResultOK    = iota // 0 - OK
+	OpResultERROR        // 1 - ERROR
+)
+
+const (
+	opStopPettingTimeoutSeconds = 10
+)
+
+// currentOpts is current operating parameters for the daemon.
+//
+// It is assigned at the first call of Run and updated on each subsequent call of it.
+var currentOpts *DaemonOpts
```

Comment:
Can you put these global variables into a struct called `WatchdogDaemon`? And a `func New() *WatchdogDaemon { ... }` function to initialize them.

Reply:
Yes, that is better idea! -- Done

Used `Daemon` as current package is `Watchdogd`. Moved watchdogd operations under the struct as well.